### PR TITLE
Remove deprecated ``qiskit.algorithms`` module

### PR DESCRIPTION
**Pull request by ElePT**
### Summary
This PR partially addresses #10991 by removing `qiskit.algorithms` and `test.python.algorithms`, updating the related sphinx references, and removing the API reference entry. 

### Details and comments

This is PR # 2 of the *"Opflow & Algorithms deprecation series"*, it is on hold until #11099  is merged, and blocks the opflow removal PR (#11111).

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@ Qiskit |version| documentation
 ##############################
 
 Qiskit is open-source software for working with quantum computers
-at the level of circuits, pulses, and algorithms.
+at the level of circuits and pulses.
```
Comment (jakelishman) [inline docs/index.rst:6]: 😢

**Review by jakelishman — COMMENTED** [review]
Thanks for doing this, Elena.  Do you know how much work is needed to fix Neko in all the relevant places before we merge this?  Technically Neko isn't part of the branch protections (iirc), but it'd still be better to have all downstream projects migrated to `qiskit_algorithms` before we remove this.

File: docs/index.rst (hunk 1 above)
Comment (ElePT) [reply docs/index.rst:6]: The "fix" was merged 10 mins ago https://github.com/Qiskit/qiskit-neko/pull/34. I've re-triggered the CI to see if it passes now.

File: releasenotes/notes/remove-qiskit-algorithms-a43541fe24b72208.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+upgrade:
+  - |
+    The ``qiskit.algorithms`` module has been removed, following its deprecation in
+    Qiskit 0.44. The code has been migrated to a standalone library (``qiskit_algorithms``)
+    and can be found on PyPi or `GitHub <https://github.com/qiskit-community/qiskit-algorithms>`_.
+    The decision to migrate the algorithms module to a separate package
+    was made to clarify the purpose Qiskit and make a distinction between the tools
+    and libraries built on top of it.
```
Comment (woodsp-ibm) [inline releasenotes/notes/remove-qiskit-algorithms-a43541fe24b72208.yaml:9]: Not all qiskit.algorithms was migrated. Only the algorithm code that uses primitives. I think it was mentioned in prior release notes that if any of opflow/Quantum instance based algorithm function was still being used a user would first need to migrate to the primitive based algorithms. Should we briefly mention this aspect again here? Or perhaps minimally change `The code has been migrated...` to `The primitive based algorithms from this code have been migrated...` or something like that? though perhaps that's less clear if you had not been paying attention and things still worked!

**Comment by woodsp-ibm** [comment]
What is the intent to deal with `qiskit/utils/validation.py`. That was used by algorithms and some apps (not sure about opflow). Maybe it needs to stay for 0.46 given it was overlooked when the algorithms&opflow were initially deprecated. It presumably does vanish for 1.0 which this PR is labelled for.

**Comment by ElePT** [comment]
> What is the intent to deal with `qiskit/utils/validation.py`. That was used by algorithms and some apps (not sure about opflow). Maybe it needs to stay for 0.46 given it was overlooked when the algorithms&opflow were initially deprecated. It presumably does vanish for 1.0 which this PR is labelled for.

I was planning to remove the deprecated utils (validation and algorithm globals) in a follow-up PR, I don't remember the exact reason why I didn't do it all at once, but I remember trying and concluding that it would be difficult before the opflow removal. So the plan is to merge this PR, then #11111 and finally the PR deleting these utils you mentioned. These removals are targeted for `1.0`, and the deprecated code will remain issuing a warning in`0.46`

File: releasenotes/notes/remove-qiskit-algorithms-a43541fe24b72208.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+upgrade:
+  - |
+    The ``qiskit.algorithms`` module has been removed, following its deprecation in
+    Qiskit 0.44. The code has been migrated to a standalone library (``qiskit_algorithms``)
```
Comment (ElePT) [inline releasenotes/notes/remove-qiskit-algorithms-a43541fe24b72208.yaml]:
```suggestion
    Qiskit 0.44. The primitive-based algorithms from this module have been migrated to a standalone library (``qiskit_algorithms``)
```

**Review by jakelishman — APPROVED** [review]
Thanks for shepherding everything else through this, Elena!

**Pull request merged**